### ISSUE TITLE
[DESIGN] user experience

### ISSUE THREAD

**Issue by justheuristic** [bigscience-workshop/petals#5]
__This issue__ is about how to make distributed bloom user-friendly. See comments below for interface prototype.

__
A bunch of things that we may want to make alongside the tech so that people can... you know... use it :)

- basic website (see training-transformers-together.github.io/ )
  - what to expect from it?
    - like openai api
  - why can't i just use ram/ssd offload?
    - it's faster together; passing activations for large models is faster than swapping layers
  - legal disclaimer: by using our model you agree to follow the model's license, read more here

- inference notebook
  - as responsive as possible, print words as they appear
  - option to show what happens under the hood (i.e. running through this guy; that guy disconnected; waiting for this)
  - [optional] make it easy to customize sampling method

- prompt-tuning notebook
  - pick some task that can be solved relatively quickly with visible results
  - TODO ask @artek0chumak on best practices
  - [optional] explain how to add other adapters / ptuning modes
  - [optional] inference prompt-tuned model? push adapters to HF?
  - [optional] collaborative prompt-tuning?

- inference website
  - the default option can simply run inference notebook on server hosted by us

- System Demonstration
  - August 1st: https://2022.emnlp.org/calls/System_Demonstrations/

**Comment by bhack** [comment]
> If we ever want to let non-programmers use the model, we can set it up using gradio via HF spaces. It is cpu-only, but the CPU it provides is better than in colab.

It would be nice (in a distant day) to simply join through the browser directly in a Federated setup:
https://github.com/epfml/disco

**Comment by justheuristic** [comment]
I'm a big fan of that framework, since it was called DeAI. However, it will take some time for us to sort through more immediate issues before we can run directly in browser

**Comment by justheuristic** [comment]
The current interface has been fully implemented, will start a new issue later
